src/AlgebraicGeometry/Schemes/CoveredSchemes/Objects/Properties.jl

```diff
@@ -1,29 +1,112 @@
-export is_empty
+export is_empty, is_smooth, is_reduced, is_integral, is_irreducible
 
 ########################################################################
 # Properties of AbsCoveredSchemes                                      #
+#    by referral to the patches                                        #
 ########################################################################
+
+########################################################################
+# (1) Check and store, whether a covered scheme is empty               #
+########################################################################
+@doc Markdown.doc"""
+   is_empty(X::AbsCoveredScheme)
+
+Return the boolean value whether a covered scheme `X` is empty.
+
+"""
 @attr function is_empty(X::AbsCoveredScheme)
   if !isdefined(X, :coverings) 
     return true
   end
   return all(x->isempty(x), all_patches(default_covering(X)))
 end
 
+########################################################################
+# (2) Check and store, whether a covered scheme is smooth              #
+########################################################################
+@doc Markdown.doc"""
+   is_smooth(X::AbsCoveredScheme)
+
+Return the boolean value whether a covered scheme `X` is smooth.
+
+"""
 @attr function is_smooth(X::AbsCoveredScheme)
   if !isdefined(X, :coverings) 
     return true
   end
-  # TODO: This can be optimized! We only need to check the rank of 
+  # TODO: Can this be optimized? We only need to check the rank of
   # the jacobian matrices where we haven't checked in another chart before. 
+  # This is a tradeoff between cost of Jacobian criterion and cost of
+  # optimizing the use of the covering
   return all(x->is_smooth(x), affine_charts(X))
 end
 
+########################################################################
+# (3) Check and store, whether a covered scheme is integral            #
+#     i.e. irreducible and reduced                                     #
+########################################################################
+@doc Markdown.doc"""
+   is_integral(X::AbsCoveredScheme)
+
+Return the boolean value whether a covered scheme `X` is integral.
+
+"""
 @attr function is_integral(X::AbsCoveredScheme)
-  return is_connected(X) && all(U->(is_integral(U)), affine_charts(X))
+  !is_empty(X) || return false
+  return is_reduced(X) && is_irreducible(X)
 end
 
+# auxilliary function for connectedness of glueing graph
+#      do not confuse with connectedness of the scheme
+# Note: This does not work with glueing_graph, because empty patches
+#      need to be ignored without changing the covering
+@attr function is_connected_glueing(X::AbsCoveredScheme)
```

Comment:
please provide a docstring including a definition of this property.

Reply:
I did not intend to publish this function, it is more for internal use.

Reply:
The docstring is not for the user. It is for me so that I may understand what this function does. 
You can also do that as comments in the code, but a docstring is accessible from the repl and comments are not.

Reply:
docstring done